### PR TITLE
(Linux) Socket stats

### PR DESCRIPTION
**Pull request by rlupton20**
Hopefully this is the last one :)

Added basic socket statistics for the system (used tcp/udp/tcp6/udp6 sockets, and orphaned tcp sockets). Plausible for BSD also (I think).

### CODE REVIEW

**Comment by rlupton20** [comment]
Also, I'm assuming you can squash commits if you like? I can rebase if that's not possible.

File: examples/info.rs [hunk 1]
```diff
@@ -73,4 +73,9 @@ fn main() {
         Ok(cpu_temp) => println!("\nCPU temp: {}", cpu_temp),
         Err(x) => println!("\nCPU temp: {}", x)
     }
+
+    match sys.socket_stats() {
+        Ok(stats) => println!("System socket statistics: {:?}", stats),
```
Comment (myfreeweb) [inline examples/info.rs]: `"\nSystem socket statistics: {:?}"`

**Comment by myfreeweb** [comment]
I can just merge, but it's always better to rebase

**Comment by rlupton20** [comment]
OK - I'll rebase these into one commit for you (I thought there was an option to squash on merge)

**Comment by rlupton20** [comment]
Bloody hell, made heavy weather of that!

All good now.

**Comment by myfreeweb** [comment]
Yeah, fine. But in general — I don't want *everything* squashed into one commit! Just always rebase on top of master, and only squash what makes sense to squash.

**Comment by rlupton20** [comment]
(I removed the commits that didn't make sense to squash during rebase, since I'd muddled my history slightly)

**Pull request merged**